lib/contractWrapperBase.ts

```diff
@@ -168,7 +168,7 @@ export abstract class ContractWrapperBase {
    * @param avatarAddress
    */
   protected async _getSchemePermissions(avatarAddress: Address): Promise<SchemePermissions> {
-    const permissions = (await this.getController(avatarAddress))
+    const permissions = await (await this.getController(avatarAddress))
```

Comment:
why the double `await`?

Reply:
there are two async functions being called here.  the bug is that I wasn't awaiting the second one

Reply:
But when you return a promise from a `then` callback, it flattens it into a single promise. Ther'e can't be a promise of a promise (`Promise<Promise<T>>`).

Reply:
The parens ensure that it is not a promise of a promise.  See also the test I added that confirms that the function works now.

Reply:
single `await` should work just as well as double `await`.

Reply:
It doesn't because you end up getting a promise from the second async call instead of the actual value being promised.